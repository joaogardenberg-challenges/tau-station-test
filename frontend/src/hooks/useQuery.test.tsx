import { render, screen } from '@testing-library/react'

const useLocation = jest.fn()

jest.doMock('react-router-dom', () => ({ useLocation }))

const useQuery = require('./useQuery').default

const Component = () => {
  const query = useQuery()

  return (
    <>
      <p>{query.get('width')}</p>
      <p>{query.toString()}</p>
    </>
  )
}

describe('useQuery', () => {
  const width = 400
  const search = `?width=${width}`

  beforeEach(() => {
    useLocation.mockReturnValue({ search })
  })

  it('functions properly in a component', () => {
    const { container } = render(<Component />)

    const widthText = screen.getByText(width.toString())
    const queryText = screen.getByText(search.substring(1))

    expect(widthText).toBeInTheDocument()
    expect(queryText).toBeInTheDocument()
    expect(container).toMatchSnapshot()
  })
})
