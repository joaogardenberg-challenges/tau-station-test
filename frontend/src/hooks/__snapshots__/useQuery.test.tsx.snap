// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useQuery functions properly in a component 1`] = `
<div>
  <p>
    400
  </p>
  <p>
    width=400
  </p>
</div>
`;
